src/HCubature.jl

```diff
@@ -46,7 +46,7 @@ end
 cubrule(::Val{0}, ::Type{T}) where {T} = Trivial()
 countevals(::Trivial) = 1
 
-function hcubature_(f, a::SVector{n,T}, b::SVector{n,T}, norm, rtol_, atol, maxevals, initdiv) where {n, T<:AbstractFloat}
+function hcubature_(f, a::SVector{n,T}, b::SVector{n,T}, norm, rtol_, atol, maxevals, initdiv) where {n, T<:Real}
```

Comment:
This will cause problems because it will circumvent the `float`  promotion [here](https://github.com/JuliaMath/HCubature.jl/blob/7f27799e7d4630480cc4c257239a80a8ed8af4cd/src/HCubature.jl#L126).

It probably should suffice to declare the parameters as `a′` and `b′` and use `T′` in the signature.  Then do `a, b = float(a′), float(b′)` as the first line followed by `T = eltype(a)`.

Reply:
or just combine that promotion method with this one.